### PR TITLE
Add availability indicator to mentor page

### LINKED ISSUE #13: Availability Indicator
**Feature Request**
Add a "status color" next to a mentor's name to indicate if they have availability set. Would help me check availability quicker rather than having to click on 21 people individually.
Example: https://gyazo.com/419e950104a7ac74ee32d85da12ccd4c

### PR DESCRIPTION
**Pull request by Evanston09**
This PR will:
- Allow staff to view if mentors have availability at a glance.
- Close #13

### CODE REVIEW

File: src/routes/(authed)/dash/mentors/+page.server.ts [hunk 1]
```diff
@@ -12,12 +12,22 @@ export const load: PageServerLoad = async ({ cookies }) => {
...
 		redirect(307, '/schedule');
 	}
-
+	const mentors = await db
+		.select()
+		.from(users)
+		.where(or(gte(users.role, ROLE_MENTOR), gte(users.roleOverride, ROLE_MENTOR)));
+	const mentorsWithAvail = mentors.map((mentor: typeof users.$inferSelect) => {
+		let isAvailable = false;
+		const mentorAvailability = JSON.parse(mentor.mentorAvailability!);
```
Comment (c0repwn3r) [inline src/routes/(authed)/dash/mentors/+page.server.ts:21]: This will return 500 on mentors that do not have availability set (which IS possible)

**Review by c0repwn3r — CHANGES_REQUESTED** [review]

File: src/routes/(authed)/dash/mentors/+page.server.ts [hunk 2]
```diff
@@ -12,12 +12,22 @@ export const load: PageServerLoad = async ({ cookies }) => {
...
+		if (mentorAvailability) {
+			isAvailable = Object.keys(mentorAvailability)
+				.filter((day) => day !== 'timezone' && day !== 'exceptions')
+				.some((day) => mentorAvailability[day].available === true);
```
[the remark this answers is not in the document]
Comment (Evanston09) [reply src/routes/(authed)/dash/mentors/+page.server.ts:25]: I think if you set availability then later get rid of it the db will show all the days of the week just the available key is set to false. So just setting it based on null won't always work

File: src/routes/(authed)/dash/mentors/+page.server.ts (hunk 1 above)
Comment (Evanston09) [reply src/routes/(authed)/dash/mentors/+page.server.ts:21]: Can you elaborate? I do not get a 500 even if the `mentorAvailability` field is null.

File: src/routes/(authed)/dash/mentors/+page.server.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (c0repwn3r) [reply src/routes/(authed)/dash/mentors/+page.server.ts:25]: Got it. In that case, please update the description at the top of the page - maybe "Green light indicates mentor is available at least one day"

File: src/routes/(authed)/dash/mentors/+page.server.ts (hunk 1 above)
Comment (c0repwn3r) [reply src/routes/(authed)/dash/mentors/+page.server.ts:21]: Ah, I didn't see this was in a .map(). Works ok

File: src/routes/(authed)/dash/mentors/+page.svelte [hunk 3]
```diff
@@ -15,7 +15,8 @@
 </script>
 
 <p class="ml-1">
-	Mentors and instructors are added automatically by their VATUSA role. You can also
+	Mentors and instructors are added automatically by their VATUSA role. Mentors with availability
```
Comment (c0repwn3r) [inline src/routes/(authed)/dash/mentors/+page.svelte:18]:
```diff
- Mentors with availability
+ Mentors available on at least one day
```
Maybe something like this, so it's more clear what the green dot indicates.

**Review by c0repwn3r — APPROVED** [review]
Good work! Bump the patch version to 1.1.1 then merge.

**Pull request merged**